### PR TITLE
[Feature] Redesigned Spark application entity

### PR DESCRIPTION
**Pull request by lenoxzhao**
## Brief change log

1. Redesigned the Spark application entity class `SparkApplication`and persisted some scheduling parameters for easier user access, such as **spark.driver.cores** and **spark.executor.cores**.
```
/** scheduling */
private String driverCores;
private String driverMemory;
private String executorCores;
private String executorMemory;
private String executorMaxNums;

/** metrics of running job */
private Long numTasks;
private Long numCompletedTasks;
private Long numStages;
private Long numCompletedStages;
private Long usedMemory;
private Long usedVCores;
```
2. Enhanced application submission configuration settings:
  - Added support for setting Spark configurations through the configuration template(**spark-application.conf**) and custom configurations(**--conf** or **-c**), with the following priority: **custom configurations** > **configuration template** > **default configurations**.
  - Added support for setting main class entry arguments.
3. Added support for abtaining metrics dashboard of running applications for display, including the following information:
 - From Spark: number of total/completed tasks, number of total/completed stages.
 - From YARN: memory and vcores used.
4. Refactored `SqlCommandParser` based on Spark SQL syntax, introducing **org.apache.spark.sql.execution.SparkSqlParser** for SQL validation.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

**Comment by wolfboys** [comment]
cc @ChengJie1053 PTAL, thx

**Comment by ChengJie1053** [comment]
> cc @ChengJie1053 PTAL, thx

Ok, I'll review it as soon as possible

File: streampark-common/src/main/scala/org/apache/streampark/common/conf/ConfigKeys.scala [hunk 1]
```diff
@@ -65,6 +65,23 @@ object ConfigKeys {
...
+  val KEY_SPARK_DYNAMIC_ALLOCATION_ENABLED = "spark.dynamicAllocation.enabled"
+
+  val KEY_SPARK_DYNAMIC_ALLOCATION_MAX_EXECUTORS = "spark.dynamicAllocation.maxExecutors"
+
+  def KEY_SPARK_SQL(prefix: String = null): String =
+    s"${Option(prefix).getOrElse("")}sql"
+
   /** about config flink */
   def KEY_APP_CONF(prefix: String = null): String =
     s"${Option(prefix).getOrElse("")}conf"
```
Comment (ChengJie1053) [inline streampark-common/src/main/scala/org/apache/streampark/common/conf/ConfigKeys.scala:97]: Whether flink parameters can be deleted

File: streampark-common/src/main/scala/org/apache/streampark/common/conf/ConfigKeys.scala [hunk 2]
```diff
@@ -87,7 +104,7 @@ object ConfigKeys {
 
   val KEY_FLINK_TABLE_PREFIX = "flink.table."
```
Comment (ChengJie1053) [inline streampark-common/src/main/scala/org/apache/streampark/common/conf/ConfigKeys.scala:116]: Whether flink parameters can be deleted

File: streampark-console/streampark-console-service/src/main/resources/db/schema-h2.sql [hunk 3]
```diff
@@ -491,102 +491,173 @@ create table if not exists `t_yarn_queue` (
...
+);
+
+
+-- ----------------------------
+-- Table structure for t_spark_effective
+-- ----------------------------
+create table if not exists `t_spark_effective` (
+  `id` bigint generated by default as identity not null,
+  `app_id` bigint not null,
+  `target_type` tinyint not null comment '1) config 2) flink sql',
```
Comment (ChengJie1053) [inline streampark-console/streampark-console-service/src/main/resources/db/schema-h2.sql]: Change flink sql to spark sql

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/SparkApplication.java [hunk 4]
```diff
@@ -251,37 +222,50 @@ public class SparkApplication extends BaseEntity {
...
         if (!(SparkExecutionMode.YARN_CLIENT == this.getSparkExecutionMode()
             || SparkExecutionMode.YARN_CLUSTER == this.getSparkExecutionMode())) {
             return;
         }
+        if (StringUtils.isBlank(this.yarnQueue)) {
+            this.yarnQueue = "default";
+        }
+        Map<String, String> queueLabelMap = YarnQueueLabelExpression.getQueueLabelMap(this.yarnQueue);
+        this.setYarnQueueName(queueLabelMap.getOrDefault(ConfigKeys.KEY_YARN_APP_QUEUE(), "default"));
+        this.setYarnQueueLabel(queueLabelMap.getOrDefault(ConfigKeys.KEY_YARN_APP_NODE_LABEL(), null));
```
Comment (ChengJie1053) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/SparkApplication.java:244]: What's the difference between `yarnQueue` and `yarnQueueName`

File: streampark-spark/streampark-spark-client/streampark-spark-client-core/src/main/scala/org/apache/streampark/spark/client/impl/YarnClient.scala [hunk 5]
```diff
@@ -119,24 +119,40 @@ object YarnClient extends SparkClientTrait {
...
     }
   }
 
+  protected def setYarnQueue(submitRequest: SubmitRequest): Unit = {
+    if (submitRequest.hasExtra("yarnQueueName")) {
+      submitRequest.appProperties.put("spark.yarn.queue", submitRequest.getExtra("yarnQueueName").asInstanceOf[String])
+    }
+    if (submitRequest.hasExtra("yarnQueueLabel")) {
+      submitRequest.appProperties.put("spark.yarn.am.nodeLabelExpression", submitRequest.getExtra("yarnQueueLabel").asInstanceOf[String])
+      submitRequest.appProperties.put("spark.yarn.executor.nodeLabelExpression", submitRequest.getExtra("yarnQueueLabel").asInstanceOf[String])
```
Comment (ChengJie1053) [inline streampark-spark/streampark-spark-client/streampark-spark-client-core/src/main/scala/org/apache/streampark/spark/client/impl/YarnClient.scala]: Whether yarnQueueName and yarnQueueLabel should be extracted as constants

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/application/impl/SparkApplicationActionServiceImpl.java [hunk 6]
```diff
@@ -291,39 +285,43 @@ public void start(SparkApplication appParam, boolean auto) throws Exception {
...
-            || SparkExecutionMode.YARN_CLIENT == application.getSparkExecutionMode()) {
-            buildResult = new ShadedBuildResponse(null, flinkUserJar, true);
-            application.setJobManagerUrl(YarnUtils.getRMWebAppURL(true));
+        if (SparkExecutionMode.isYarnMode(application.getSparkExecutionMode())) {
+            buildResult = new ShadedBuildResponse(null, sparkUserJar, true);
+            if (StringUtils.isNotBlank(application.getYarnQueueName())) {
+                extraParameter.put("yarnQueueName", application.getYarnQueueName());
+            }
+            if (StringUtils.isNotBlank(application.getYarnQueueLabel())) {
+                extraParameter.put("yarnQueueLabel", application.getYarnQueueLabel());
```
Comment (ChengJie1053) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/application/impl/SparkApplicationActionServiceImpl.java]: Whether yarnQueueName and yarnQueueLabel should be extracted as constants

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/application/impl/SparkApplicationManageServiceImpl.java [hunk 7]
```diff
@@ -294,90 +283,83 @@ public boolean create(SparkApplication appParam) {
...
+        newApp.setRestartSize(oldApp.getRestartSize());
+        newApp.setState(SparkAppStateEnum.ADDED.getValue());
+        newApp.setOptions(oldApp.getOptions());
+        newApp.setOptionState(OptionStateEnum.NONE.getValue());
+        newApp.setUserId(ServiceHelper.getUserId());
+        newApp.setDescription(oldApp.getDescription());
+        newApp.setRelease(ReleaseStateEnum.NEED_RELEASE.get());
+        newApp.setAlertId(oldApp.getAlertId());
+        newApp.setCreateTime(new Date());
+        newApp.setModifyTime(new Date());
```
Comment (ChengJie1053) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/application/impl/SparkApplicationManageServiceImpl.java]: Whether the same Date object can be used

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/SparkApplication.java (hunk 4 above)
Comment (lenoxzhao) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/entity/SparkApplication.java:244]: According to this [document](https://streampark.apache.org/docs/platform/yarn-queue), `yarnQueue` would be passed in format like `queue@label1,label2`. `YarnQueueLabelExpression` can parse queue name and queue label from `yarnQueue`. So `yarnQueueName` is used to represtented "queue" and `yarnQueueLabel` is used to represent "label1, label2".

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/application/impl/SparkApplicationManageServiceImpl.java (hunk 7 above)
Comment (lenoxzhao) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/application/impl/SparkApplicationManageServiceImpl.java]: done.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/application/impl/SparkApplicationActionServiceImpl.java (hunk 6 above)
Comment (lenoxzhao) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/application/impl/SparkApplicationActionServiceImpl.java]: Already extracted as constants in `org.apache.streampark.common.conf.ConfigKeys`.

File: streampark-spark/streampark-spark-client/streampark-spark-client-core/src/main/scala/org/apache/streampark/spark/client/impl/YarnClient.scala (hunk 5 above)
Comment (lenoxzhao) [reply streampark-spark/streampark-spark-client/streampark-spark-client-core/src/main/scala/org/apache/streampark/spark/client/impl/YarnClient.scala]: Already extracted as constants in `org.apache.streampark.common.conf.ConfigKeys`.

File: streampark-console/streampark-console-service/src/main/resources/db/schema-h2.sql (hunk 3 above)
Comment (lenoxzhao) [reply streampark-console/streampark-console-service/src/main/resources/db/schema-h2.sql]: Done.

**Comment by ChengJie1053** [comment]
Please fix Some checks were not successful

**Review by ChengJie1053 — APPROVED** [review]
LGTM

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**